airflow/providers/google/cloud/operators/cloud_build.py

```diff
@@ -15,121 +15,933 @@
+    :type project_id: Optional[str]
+    :param wait: Optional, wait for operation to finish.
+    :type wait: Optional[bool]
+    :param retry: Optional, a retry object used  to retry requests. If `None` is specified, requests
+        will not be retried.
+    :type retry: Optional[Retry]
+    :param timeout: Optional, the amount of time, in seconds, to wait for the request to complete.
+        Note that if `retry` is specified, the timeout applies to each individual attempt.
+    :type timeout: Optional[float]
+    :param metadata: Optional, additional metadata that is provided to the method.
+    :type metadata: Optional[Sequence[Tuple[str, str]]]
+    :param gcp_conn_id: Optional, the connection ID used to connect to Google Cloud Platform.
+    :type gcp_conn_id: Optional[str]
+    :param impersonation_chain: Optional service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account (templated).
+    :type impersonation_chain: Union[str, Sequence[str]]
+
+    :rtype: dict
+    """
+
+    template_fields = ("project_id", "build", "body", "gcp_conn_id", "impersonation_chain")
+
+    def __init__(
+        self,
+        *,
+        build: Union[Dict, Build, str],
+        body: Optional[Dict] = None,
+        project_id: Optional[str] = None,
+        wait: bool = True,
+        retry: Optional[Retry] = None,
+        timeout: Optional[float] = None,
+        metadata: Optional[Sequence[Tuple[str, str]]] = None,
+        gcp_conn_id: str = "google_cloud_default",
+        impersonation_chain: Optional[Union[str, Sequence[str]]] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.build = build
+        # Not template fields to keep original value
+        self.build_raw = build
+        self.body = body
+        self.project_id = project_id
+        self.wait = wait
+        self.retry = retry
+        self.timeout = timeout
+        self.metadata = metadata
+        self.gcp_conn_id = gcp_conn_id
+        self.impersonation_chain = impersonation_chain
+
+        if self.body:
+            warnings.warn(
+                "The body parameter has been deprecated. You should pass body using " "the build parameter.",
+                DeprecationWarning,
+                stacklevel=4,
+            )
+            if not self.build:
+                self.build = self.build_raw = self.body
+
+    def prepare_template(self) -> None:
```

Comment:
```suggestion
    def _prepare_template(self) -> None:
```
Or is this method part of public interface?

Reply:
This one is to override the prepare_template in the baseoperator.py. I agree it's more like an internal function. We can refactor it in another PR instead of this PR. WDYT?

Reply:
Sounds good to me